### PR TITLE
March 2023

### PR DESCRIPTION
**Pull request by jprix**
updated tail wind and yaml versions
added:  create Allocation builder, viewing Allocations by Id
bunch of minor code optimizations and improvements.

### CODE REVIEW

File: src/UI/Orders/ListOrders.svelte [hunk 1]
```diff
@@ -61,15 +61,15 @@
...
-  const updateOrders = async () => {
-    try {
-      const filteredOrders = await getOrders(stringifiedQueryParams);
-      ordersFilter = filteredOrders.orders;
-      ordersFilterView = true;
-    } catch (e) {
-      alert(e);
-    }
-  };
+  // const updateOrders = async () => {
```
Comment (rcbgr) [inline src/UI/Orders/ListOrders.svelte:64]: Remove unused code?

File: src/UI/Orders/ListOrders.svelte [hunk 2]
```diff
@@ -92,20 +92,17 @@
...
 
-    const filteredQueryParams = _.omitBy(
-      queryParams,
-      (v) => _.isUndefined(v) || _.isNull(v) || v === ''
-    );
-
-    stringifiedQueryParams = queryString.stringify(filteredQueryParams);
-    closeForm();
-    await updateOrders();
+     const stringifiedQueryParams = generateQueryparams(queryParams);
```
Comment (rcbgr) [inline src/UI/Orders/ListOrders.svelte:95]: Fix indention

File: src/apis/Allocations.js [hunk 3]
```diff
@@ -14,17 +14,33 @@
...
+  }
+};
+
 export const getAllocations = async (queryParams) => {
+  
   const { port, portfolioId, httpHost } = await fetchStore();
   const path = `/v1/portfolios/${portfolioId}/allocations`;
   const allocationsUrl = `${httpHost}:${port}/api${path}?${
-    queryParams ? queryParams : 'start_date=2023-01-05T00:00:01Z'
+    queryParams ? queryParams : 'start_date=2023-04-05T00:00:01Z'
```
Comment (rcbgr) [inline src/apis/Allocations.js:43]: Should this be now - N months?

File: src/apis/Exchange/Orders.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  } catch (e) {
+    return e;
+  }
+};
+
+export const getOrderById = async () => {
+  const { port, httpHost } = await fetchStore();
+
+  const url = `${httpHost}:${port}/api/exchange/orders/9a5cd33b-e371-4ff0-a7f1-5804250e38cc`;
+  const path = "/orders/9a5cd33b-e371-4ff0-a7f1-5804250e38cc";
```
Comment (rcbgr) [inline src/apis/Exchange/Orders.js:34]: What are these portfolio ids? Make an env var?

File: src/apis/Orders.js [hunk 5]
```diff
@@ -137,7 +137,7 @@ export const getOrders = async (queryParams) => {
   const { port, portfolioId, httpHost } = await fetchStore();
   const path = `/v1/portfolios/${portfolioId}/orders`;
   const ordersUrl = `${httpHost}:${port}/api/v1/portfolios/${portfolioId}/orders?${
-    queryParams ? queryParams : 'start_date=2022-06-05T00:00:01Z'
+    queryParams ? queryParams : 'start_date=2023-03-05T00:00:01Z'
```
Comment (rcbgr) [inline src/apis/Orders.js:140]: Change to now - N months?

File: src/routes/Orders/CreateAllocation.svelte [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  let payload = [];
+  let selectedIds = [];
+  let showOrdersSelect = false;
+  let showOrdersSideSelect = false;
+
+  let selectedAsset;
+  let selectedSide;
+
+
+  // onMount(async () => {
```
Comment (rcbgr) [inline src/routes/Orders/CreateAllocation.svelte:28]: Remove unused code

File: src/routes/Orders/CreateAllocation.svelte [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+ const generateOrdersList = async (queryParams) => {
+  //72 hours time cap
```
Comment (rcbgr) [inline src/routes/Orders/CreateAllocation.svelte:31]: What does this mean?

File: src/routes/Orders/CreateAllocation.svelte [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+ console.log(startDate);
+  
+  const alertPrompt = (result) => {
+    alert('Allocation initiated', result);
+     navigate('/Orders/Allocations');
+  }
+
+   const handleAssetSelect = async () => {
+    // selectedAsset = event.target.value;
+   const queryParams=  {
```
Comment (rcbgr) [inline src/routes/Orders/CreateAllocation.svelte:54]: Formatting is off

File: tailwind.config.js [hunk 9]
```diff
@@ -1,17 +1,27 @@
...
+const production = !process.env.ROLLUP_WATCH;
+
 module.exports = {
   future: {
     purgeLayersByDefault: true,
     removeDeprecatedGapUtilities: true,
   },
   plugins: [
-
+    // ...
```
Comment (rcbgr) [inline tailwind.config.js:9]: Remove empty comment

**Review by rcbgr — CHANGES_REQUESTED** [review]

File: src/apis/Allocations.js (hunk 3 above)
Comment (jprix) [reply src/apis/Allocations.js:43]: fixed it to go back N # of days automatically.

File: src/apis/Exchange/Orders.js (hunk 4 above)
Comment (jprix) [reply src/apis/Exchange/Orders.js:34]: i actually threw that in to test something for a customer today..  i just removed the code, its not needed for the app.

File: src/routes/Orders/CreateAllocation.svelte (hunk 7 above)
Comment (jprix) [reply src/routes/Orders/CreateAllocation.svelte:31]: that was a note to myself.  you can only allocate orders < 72 hours old.

File: src/utils/constants.js [hunk 10]
```diff
@@ -1,11 +1,43 @@
...
+  date.setDate(date.getDate() - daysAgo);
+
+  const year = date.getUTCFullYear();
+  const month = ('0' + (date.getUTCMonth() + 1)).slice(-2);
+  const day = ('0' + date.getUTCDate()).slice(-2);
+  const hours = ('0' + date.getUTCHours()).slice(-2);
+  const minutes = ('0' + date.getUTCMinutes()).slice(-2);
+  const seconds = ('0' + date.getUTCSeconds()).slice(-2);
+
+  return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}Z`;
```
Comment (cfluke-cb) [inline src/utils/constants.js:28]: why not use toISOString? 
let d = new Date()
d.toISOString()
'2023-05-01T18:01:42.026Z'

File: src/routes/Orders/CreateAllocation.svelte [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+    for="choose-asset"><b>Asset:</b></label>
+    <br />
+  <select
+     id="choose-asset" 
+     name="choose-asset"
+     class="focus:outline-none focus:shadow-outline mb-3 w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow"
+     bind:value={selectedAsset} 
+     on:change={handleAssetSelect}>
+  <option value="">Choose Asset</option>
+  <option value="BTC-USD">BTC-USD</option>
```
Comment (cfluke-cb) [inline src/routes/Orders/CreateAllocation.svelte:101]: I would make this a constant array and share it other places this is used.

Comment (cfluke-cb) [reply src/routes/Orders/CreateAllocation.svelte:101]: like on [PlaceExchangeOrder.svelte](https://github.com/coinbase-samples/prime-explorer-js/pull/17/files#diff-11de12c3afd38e5e64a5e2cab88491c51971018c1c34d0ec6f72d022793d83fa)

File: src/routes/Exchange/Orders/ExchangeOrderId.svelte [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+  import { getOrderById } from '../../../apis/Exchange/Orders';
+  import { Tile, CodeSnippet, Content } from 'carbon-components-svelte';
+  import Nav from '../../../Nav.svelte';
+  
+  import { onMount } from 'svelte';
+
+  let orderIdDetails;
+  let payload;
+  let open = false;
+  let date, user_id, product_id, side, type, status;
```
Comment (cfluke-cb) [inline src/routes/Exchange/Orders/ExchangeOrderId.svelte:28]: any reason to be mixing individual lines to init, vs commas?

File: src/routes/Exchange/Orders/ExchangeOpenOrders.svelte [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+        ><br />
+        <select
+          id="product_id"
+          name="product_id"
+          class="focus:outline-none focus:shadow-outline mb-3 w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow"
+          on:change={handleChange}
+          bind:value={$form.product_id}
+        >
+          <option />
+          <option>BTC-USD</option>
```
Comment (cfluke-cb) [inline src/routes/Exchange/Orders/ExchangeOpenOrders.svelte:120]: share array as constant

File: src/routes/Exchange/Orders/ExchangeFills.svelte [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+      class="focus:outline-none focus:shadow-outline rounded bg-blue-500 py-2 px-4 font-bold text-white hover:bg-blue-700"
+      >Filter Orders</Button
+    >
+  {/if}
+ 
+  {#if ordersFilterView} 
+     <DataTable
+      size="short"
+      headers={[
+        { key: 'id', value: 'Order Id' },
```
Comment (cfluke-cb) [inline src/routes/Exchange/Orders/ExchangeFills.svelte:153]: same copy/paste as [ExchangeOpenOrders.svelte](https://github.com/coinbase-samples/prime-explorer-js/pull/17/files#diff-0cc25b50b61b095b83d800bac8975075d9ab05781722180c882dd731c15e48df)

File: src/UI/WalletBalance.svelte [hunk 15]
```diff
@@ -19,14 +19,21 @@
...
 </script>
 
-<h5>{balance}</h5>
+<h4>Balance Amount: {walletBalanceDetails?.balance.amount}</h4>
+<h4>Bonded Amount: {walletBalanceDetails?.balance.bonded_amount}</h4>
+<h4>Reserved Amount: {walletBalanceDetails?.balance.reserved_amount}</h4>
+<h4>Unbonded Amount: {walletBalanceDetails?.balance.unbonding_amount}</h4>
+<h4>Unvested Amount: {walletBalanceDetails?.balance.unvested_amount}</h4>
+<h4>Rewards Amount: {walletBalanceDetails?.balance.pending_rewards_amount}</h4>
+<h4>Past Rewards Amout: {walletBalanceDetails?.balance.past_rewards_amount}</h4>
```
Comment (cfluke-cb) [inline src/UI/WalletBalance.svelte:34]: typo

File: src/routes/Orders/CreateAllocation.svelte [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+  <select
+  id="choose-side" 
+  name="choose-side"
+  class="focus:outline-none focus:shadow-outline mb-3 w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow"
+  bind:value={selectedSide}
+>
+  <option value="ss">Choose Order Side</option>
+  <option value="BUY">BUY</option>
+  <option value="SELL">SELL</option>
+</select>
```
Comment (cfluke-cb) [inline src/routes/Orders/CreateAllocation.svelte:87]: have you run npm run lint and npm run format?

File: src/routes/Orders/CreateAllocation.svelte (hunk 11 above)
Comment (jprix) [reply src/routes/Orders/CreateAllocation.svelte:130]: great point. was meaning to get around to that. done.

File: src/utils/constants.js [hunk 17]
```diff
@@ -1,11 +1,62 @@
...
+  'REN-USD',
+  'UMA-USD',
+  '1INCH-USD',
+  'ALCX-USD',
+  'BNT-USD',
+  'CEL-USD',
+  'ENJ-USD',
+  'GRT-USD',
+  'KNC-USD',
+  'OMG-USD',
```
Comment (cfluke-cb) [inline src/utils/constants.js:61]: omg got delisted

File: src/utils/constants.js [hunk 18]
```diff
@@ -1,11 +1,62 @@
...
+  'SNX-USD',
+  'COMP-USD',
+  'CRV-USD',
+  'BAL-USD',
+  'REN-USD',
+  'UMA-USD',
+  '1INCH-USD',
+  'ALCX-USD',
+  'BNT-USD',
+  'CEL-USD',
```
[the remark this answers is not in the document]
Comment (jprix) [reply src/utils/constants.js:57]: that was auto-genned, i dont need em.

**Review by rcbgr — APPROVED** [review]
Please squash commits and then merge

**Review by cfluke-cb — APPROVED** [review]

**Pull request merged**